### PR TITLE
Adding Enchanted item functionality to ingredients for recipes - BUKKIT-2286

### PR DESCRIPTION
**Pull request by jwebbdev**
https://bukkit.atlassian.net/browse/BUKKIT-2286
Craft-Bukkit Pull Request: https://github.com/Bukkit/CraftBukkit/pull/859

Adds functionality that allows creating recipes that have components with specific enchantments.  This means you can now have recipes such as combining two Diamond Sword Sharpness I and creating a Diamond Sword Sharpness II, and much more.

### CODE REVIEW

**Comment by riking** [comment]
Has this been satisfied by the recent ItemMeta changes?
Also, does it have a ticket?

I don't think that any recipes I could add to vanilla with a simple mod could distinguish enchantments either, though...

File: src/main/java/org/bukkit/inventory/ShapelessRecipe.java [hunk 1]
```diff
@@ -209,4 +222,4 @@ public ItemStack getResult() {
         }
         return result;
     }
-}
+}
```
Comment (aikar) [inline src/main/java/org/bukkit/inventory/ShapelessRecipe.java:225]: Your editor has a broken configuration. All files should end with a new line, but yours stripped it. Configure it to do the opposite and add a new line if it does not exists.

File: src/main/java/org/bukkit/inventory/ShapelessRecipe.java [hunk 2]
```diff
@@ -104,6 +104,19 @@ public ShapelessRecipe addIngredient(int count, Material ingredient, int rawdata
     }
 
     /**
+     * Adds the specified ingredient.
+     *
+     * @param ingredient The ingredient to add.
+     * @return The changed recipe, so you can chain calls.
+     */
+    public ShapelessRecipe addIngredient(ItemStack item) {
+    	item.setAmount(1);
```
Comment (aikar) [inline src/main/java/org/bukkit/inventory/ShapelessRecipe.java:113]: should clone before doing this, as the plugins object will now be modified...

File: src/main/java/org/bukkit/inventory/ShapedRecipe.java [hunk 3]
```diff
@@ -99,6 +99,20 @@ public ShapedRecipe setIngredient(char key, Material ingredient, int raw) {
...
 
     /**
+     * Sets the item that a character in the recipe shape refers to.
+     *
+     * @param key The character that represents the ingredient in the shape.
+     * @param ingredient The ingredient.
+     * @return The changed recipe, so you can chain calls.
+     */
+    public ShapedRecipe setIngredient(char key, ItemStack item) {
+    	item.setAmount(1);
```
Comment (aikar) [inline src/main/java/org/bukkit/inventory/ShapedRecipe.java:109]: should clone before doing this, as the plugins object will now be modified...

File: src/main/java/org/bukkit/inventory/ShapedRecipe.java [hunk 4]
```diff
@@ -132,4 +146,4 @@ public ShapedRecipe setIngredient(char key, Material ingredient, int raw) {
     public ItemStack getResult() {
         return output.clone();
     }
-}
+}
```
Comment (aikar) [inline src/main/java/org/bukkit/inventory/ShapedRecipe.java:149]: Your editor has a broken configuration. All files should end with a new line, but yours stripped it. Configure it to do the opposite and add a new line if it does not exists.

File: src/main/java/org/bukkit/inventory/ShapelessRecipe.java [hunk 5]
```diff
@@ -104,6 +104,19 @@ public ShapelessRecipe addIngredient(int count, Material ingredient, int rawdata
     }
 
     /**
+     * Adds the specified ingredient.
+     *
+     * @param ingredient The ingredient to add.
+     * @return The changed recipe, so you can chain calls.
+     */
+    public ShapelessRecipe addIngredient(ItemStack item) {
```
Comment (riking) [inline src/main/java/org/bukkit/inventory/ShapelessRecipe.java:112]: Please rename this method (in both) to something resembling "addIngredientWithMeta", as that is the intent of your change.

**Comment by Wolvereness** [comment]
https://github.com/Bukkit/CraftBukkit/pull/859#issuecomment-21969259

**Pull request closed without merging**